sdk/storage/inc/adls/protocol/adls_rest_client.hpp

```diff
@@ -0,0 +1,2220 @@
+    uint32_t Timeout = uint32_t();  // The timeout parameter is expressed in seconds. For more information, see <a href="https://docs.microsoft.com/en-us/rest/api/storageservices/fileservices/setting-timeouts-for-blob-service-operations">Setting Timeouts for Blob Service Operations.</a>
+    std::string ApiVersionParameter = k_DefaultServiceApiVersion;  // Specifies the version of the operation to use for this request.
+  };
+
+  static Azure::Core::Http::Request ServiceListFileSystemsCreateRequest(std::string url, const ServiceListFileSystemsOptions& serviceListFileSystemsOptions)
+  {
+    Azure::Core::Http::Request request(Azure::Core::Http::HttpMethod::Get, url);
+    request.AddQueryParameter(k_QueryResource, "account");
+    if (!serviceListFileSystemsOptions.Prefix.empty())
+    {
+      request.AddQueryParameter(k_QueryPrefix, serviceListFileSystemsOptions.Prefix);
+    }
+    if (!serviceListFileSystemsOptions.Continuation.empty())
+    {
+      request.AddQueryParameter(k_QueryContinuation, serviceListFileSystemsOptions.Continuation);
+    }
+    if (!std::to_string(serviceListFileSystemsOptions.MaxResults).empty())
+    {
+      request.AddQueryParameter(k_QueryMaxResults, std::to_string(serviceListFileSystemsOptions.MaxResults));
+    }
+    if (!serviceListFileSystemsOptions.ClientRequestId.empty())
+    {
+      request.AddHeader(k_HeaderClientRequestId, serviceListFileSystemsOptions.ClientRequestId);
+    }
+    if (!std::to_string(serviceListFileSystemsOptions.Timeout).empty())
+    {
+      request.AddQueryParameter(k_QueryTimeout, std::to_string(serviceListFileSystemsOptions.Timeout));
+    }
+    request.AddHeader(k_HeaderApiVersionParameter, serviceListFileSystemsOptions.ApiVersionParameter);
+    return request;
+  }
+
+  struct ServiceListFileSystemsResponse
+  {
+    FileSystemList FileSystemList;
+    std::string Date;
+    std::string RequestId;
+    std::string Version;
+    std::string Continuation;
+    std::string ContentType;
+  };
+
+  static ServiceListFileSystemsResponse ServiceListFileSystemsParseResponse(/*const*/ std::unique_ptr<Azure::Core::Http::Response>& responsePtr)
+  {
+    ServiceListFileSystemsResponse result;
+    /* const */ auto& response = *responsePtr;
+    if (response.GetStatusCode() == Azure::Core::Http::HttpStatusCode::Ok)
+    {
+      // OK
+      result.Date = response.GetHeaders().at(k_HeaderDate);
+      result.RequestId = response.GetHeaders().at(k_HeaderXMsRequestId);
+      result.Version = response.GetHeaders().at(k_HeaderXMsVersion);
+      result.Continuation = response.GetHeaders().at(k_HeaderXMsContinuation);
+      result.ContentType = response.GetHeaders().at(k_HeaderContentType);
+      result.FileSystemList = FileSystemList::CreateFromJson(nlohmann::json::parse(response.GetBodyBuffer()));
+    }
+    else
+    {
+      std::cout << "Error occured:\n"
```

Comment:
this is okay if currently only for debug purpose.

Reply:
Yes, this is pending for the XML parser to be ready to code properly.